### PR TITLE
Update overview.md

### PR DESCRIPTION
**Pull request by eamonryan**
#### PR Description 
Updates the overview docs page to add an answer to the "Why not just use Telegraf" question.

#### Which issue(s) this PR fixes 
No GitHub issue to link, but this is asked frequently so having a place to point to would help.

#### Notes to the Reviewer
N/A

#### PR Checklist

- [NA] CHANGELOG updated 
- [X] Documentation added
- [NA] Tests updated

### CODE REVIEW

File: docs/overview.md [hunk 1]
```diff
@@ -93,6 +93,22 @@ official code with extra functionality. This allows the Agent to give an
 experience closest to its official counterparts compared to alternatives which
 may try to reimplement everything from scratch.
 
+### Why not just use Telegraf?
+Telegraf is a fantastic project and was actually considered as an alternative to 
+building a custom Prometheus scraper and agent.
```
Comment (rfratto) [inline docs/overview.md:98]: Hmm, I don't want to give the impression that we've built a custom Prometheus scraper here. Maybe we should just say something like

```
as an alternative to building our own agent
```

WDYT?

File: docs/overview.md [hunk 2]
```diff
@@ -93,6 +93,22 @@ official code with extra functionality. This allows the Agent to give an
...
+It could work, but ultimately it was not chosen due to lacking service discovery 
+and metadata label propagation.
+While these features could theoretically be added to Telegraf as OSS contributions, 
+there would be a lot of forced hacks involved due to its current design.
+
+Additonally, Telegraf is a much larger project with its own goals for its community,
+so any changes need to fit the general use cases it was designed for.
+
+With the Grafana Cloud Agent as its own project, we can deliver a more curated agent 
+specifically designed to work seamlessly with GrafanaCloud and other
```
Comment (rfratto) [inline docs/overview.md:108]:
```suggestion
specifically designed to work seamlessly with Grafana Cloud and other 
```

I think Grafana Cloud is two words, though now I'm not 100% sure :)

File: docs/overview.md [hunk 3]
```diff
@@ -93,6 +93,22 @@ official code with extra functionality. This allows the Agent to give an
...
+and metadata label propagation.
+While these features could theoretically be added to Telegraf as OSS contributions, 
+there would be a lot of forced hacks involved due to its current design.
+
+Additonally, Telegraf is a much larger project with its own goals for its community,
+so any changes need to fit the general use cases it was designed for.
+
+With the Grafana Cloud Agent as its own project, we can deliver a more curated agent 
+specifically designed to work seamlessly with GrafanaCloud and other 
+remote_write compatible Prometheus endpoints as well as Loki for logs
```
Comment (rfratto) [inline docs/overview.md:109]:
```suggestion
`remote_write` compatible Prometheus endpoints as well as Loki for logs
```

**Review by rfratto — COMMENTED** [review]
Just a few small nits, but this looks good, thanks!

File: docs/overview.md (hunk 2 above)
Comment (eamonryan) [reply docs/overview.md:108]: It's two words at https://grafana.com/docs/grafana-cloud/ so I'll say yes :D

File: docs/overview.md (hunk 1 above)
Comment (eamonryan) [reply docs/overview.md:98]: Sounds good to me!

**Review by rfratto — APPROVED** [review]
LGTM! Feel free to merge when the CI goes green :)

**Pull request merged**